staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -192,8 +200,9 @@ type CustomResourceDefinition struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// Spec describes how the user wants the resources to appear
-	Spec CustomResourceDefinitionSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Spec CustomResourceDefinitionSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
```

Comment:
A CRD with no spec is probably wrong, but is this a backward incompatible change to make an optional field required?

Reply:
it wasn't actually optional. if you submitted that object to the server, the server rejected it.